python/ray/tune/trial_runner.py

```diff
@@ -195,15 +195,17 @@ def _launch_trial(self, custom_trial=None):
             trial.start()
```

Comment:
is the nested try-except standard? seems a little odd; consider 
```syntax=python
def _launch_trial(self, custom_trial=None):
    trial = custom_trial or self._get_runnable()
    self._commit_resources(trial.resources)
    for tries in range(3):
        try:
            trial.start()
            self._running[trial.train_remote()] = trial
            break
        except Exception:
            error_msg = traceback.format_exc()
            msg = "retrying" if tries < 2 else "aborting"
            print("Error starting runner, {}:".format(msg), error_msg)
            time.sleep(2)
            trial.stop(error=True, error_msg=error_msg)
            # note that we don't return the resources, since they may
            # have been lost
```

Reply:
It's ok-ish if you only need to retry once, otherwise you'd do a loop like the above